### DISCUSSION TITLE
Getting ipynb to open as a notebook in GitHub codespace with openFiles?

### DISCUSSION THREAD

**Discussion by pamelafox** [community/community#58399] [Codespaces] [answered]
### Select Topic Area

Question

### Body

** Also posted in Jupyter extension repo: https://github.com/microsoft/vscode-jupyter/discussions/13723 **

I am trying to create a dev container that automatically opens up a particular notebook file using the Jupyter extension. All of my attempts result in the notebook rendering as JSON instead of a notebook.

Example devcontainer.json:

```
{
	"name": "Data 8: Project 1 Notebook",
	"build": {
		"dockerfile": "../dockerfile",
		"context": "../.."
	},
	"waitFor": "onCreateCommand",
	"customizations": {
		"codespaces": {
			"openFiles": [
			  "materials/fa22/project/project1/project1.ipynb"
			]
		},
		"vscode": {
			"settings": {
				"python.defaultInterpreterPath": "/usr/local/bin/python",
				"workbench.editorAssociations": {
					"*.ipynb": "jupyter-notebook"
				},
				"notebook.compactView": false
			},
			"extensions": [
				"ms-python.python",
				"ms-toolsai.jupyter",
				"MS-vsliveshare.vsliveshare"
			]
		}
	}
}
```

For this repo:
https://github.com/pamelafox/materials-fa22

Here's another repo that demonstrates the issue. If you create a Codespace, you'll notice it opens 4 ipynb files, but they render as JSON, not notebooks.

https://github.com/janosh/pymatviz

If you then close them and re-open them, they open as expected, with the notebook UI.

I'm not sure if this is an extension issue or a Codespace issue, as I assume it's related to timing of the customizations and extensions. It'd be really fantastic if there was a way to make this work, since this use case would be very popular with data science teachers.

**Comment by pamelafox** [comment]
Also filed here: https://github.com/microsoft/vscode-jupyter/discussions/13723#discussioncomment-6213993

**Comment by dmgardiner25** [comment, marked as answer]
@pamelafox I see in your other discussion that it might be an issue with VS Code core, but I'm also not sure that it's a supported scenario in Codespaces since the file will be opened before the extensions (specifically the Notebooks extension) in your codespace will be activated. One workaround that should work is to remove the `openFile` customization in your `devcontainer.json` and add `postAttachCommand: "code /workspaces/materials-fa22/materials/fa22/project/project1/project1.ipynb"`. Let me know if you are still running into the same issue after giving that a try!

**Reply by pamelafox** [reply]
Ah great workaround, thank you!

**Reply by bat52** [reply]
Thank you, I was trying to do the same and failing.
Can you help me a bit with using this workaround... I feel I am missing some background here: where do I find devcontainer.json, and where should I add postAttachCommand?

**Reply by pamelafox** [reply]
You would put devcontainer.json inside .devcontainer folder in the root of the project. It'd look like this:

```
{
    "name": "Notebook project",
    "image": "mcr.microsoft.com/devcontainers/universal:2",
    "customizations": {
        "codespaces": {
            "openFiles": []
        },
        "vscode": {
            "settings": {
                "python.defaultInterpreterPath": "/usr/local/bin/python"
            },
            "extensions": [
                "ms-python.python",
                "ms-toolsai.jupyter",
                "MS-vsliveshare.vsliveshare"
            ]
        }
    },
    "updateContentCommand": "python3 -m pip install -r requirements.txt",
    "postAttachCommand": "code /workspaces/materials-fa22/materials/fa22/hw/hw01/hw01.ipynb"
}
```

You can read more about dev containers in VS Code docs-

https://code.visualstudio.com/docs/devcontainers/containers
